athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java

```diff
@@ -0,0 +1,211 @@
+/*-
+ * #%L
+ * athena-neptune
+ * %%
+ * Copyright (C) 2019 - 2023 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.neptune.rdf;
+
+import com.amazonaws.athena.connectors.neptune.NeptuneConnection;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.neptune.auth.NeptuneSigV4SignerException;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.URI;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.vocabulary.XMLSchema;
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.QueryLanguage;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+//import org.eclipse.rdf4j.model.IRI;
+
+@SuppressWarnings("deprecation")
+public class NeptuneSparqlConnection extends NeptuneConnection 
+{
+    private static final Logger logger = LoggerFactory.getLogger(NeptuneSparqlConnection.class);
+
+    String connectionString = null;
+    boolean trimURI = false;
+    NeptuneSparqlRepository neptuneSparqlRepo = null;
+    RepositoryConnection connection = null;
+    TupleQueryResult queryResult = null;
+
+    public NeptuneSparqlConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM, String region) 
+    {
+        super(neptuneEndpoint, neptunePort, enabledIAM, region);
+        this.connectionString = "https://" + neptuneEndpoint + ":" + neptunePort;
+    }
+
+    public void connect() throws NeptuneSigV4SignerException
```

Comment:
why does this need a separate connect() class? why not include this as a part of the constructor?

Reply:
connect now from constructor